src/DefaultFileSystem.cpp

```diff
@@ -75,11 +76,35 @@ DefaultFileSystem::Read(const std::string& path) const
   return result;
 }
 
+std::string tmpfile(std::string pathnormalized)
+{
+  int poslastpoint = pathnormalized.find_last_of(".");
+  std::string filename = pathnormalized.substr(0, poslastpoint - 1);
+  std::string tempfile = filename + ".tmp";
+  return tempfile;
+}
+
+std::wstring _wtmpfile(std::wstring pathnormalized)
+{
+  int poslastpoint = pathnormalized.find_last_of(L".");
+  std::wstring filename = pathnormalized.substr(0, poslastpoint - 1);
+  std::wstring tempfile = filename + L".tmp";
+  return tempfile;
+}
+
 void DefaultFileSystem::Write(const std::string& path,
                               std::tr1::shared_ptr<std::istream> data)
-{
-  std::ofstream file(NormalizePath(path).c_str(), std::ios_base::out | std::ios_base::binary);
+{ 
+  std::ofstream file(tmpfile(NormalizePath(path)).c_str(), std::ios_base::out | std::ios_base::binary);
+  if (!file.good())
+    return;
   file << Utils::Slurp(*data);
+  if (!file.good())
+    return;
+  file.close();
+  if (!file.good())
+    return;
+  rename(tmpfile(NormalizePath(path)).c_str(), NormalizePath(path).c_str());
```

Comment:
this line is not a duplicated of function DefaultFileSystem::Move 
because  DefaultFileSystem::Move accepts only std::string ...